streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -741,18 +741,20 @@ private KafkaStreams(final InternalTopologyBuilder internalTopologyBuilder,
         final Map<Long, StreamThread.State> threadState = new HashMap<>(threads.length);
         final ArrayList<StateStoreProvider> storeProviders = new ArrayList<>();
         for (int i = 0; i < threads.length; i++) {
+            final int threadIdx = i + 1;
```

Comment:
Maybe replace `i` in the loop with `threadIdx` -- or just pass `i + 1` into `create(...)`?